cmd/fs-v1.go

```diff
@@ -110,17 +226,25 @@ func newFSObjectLayer(fsPath string) (ObjectLayer, error) {
 
 	// Load `format.json`.
 	format, err := loadFormatFS(fsPath)
-	if err != nil && err != errUnformattedDisk {
+	if err != nil && err != errUnformattedDisk && err != errFSFormatOld {
 		return nil, fmt.Errorf("Unable to load 'format.json', %s", err)
 	}
 
 	// If the `format.json` doesn't exist create one.
-	if err == errUnformattedDisk {
+	if err == errUnformattedDisk || err == errFSFormatOld {
+		// Disk is in old format migrate object metadata.
+		if err == errFSFormatOld {
+			if merr := migrateFSObjectMetadataDir(fsPath, fsUUID); merr != nil {
```

Comment:
Can we do migrate with the following style?
```
migrateFSObject()
|___ migrateFSObjectV1toV2()
```

That way we will always remember users that have real old installation.